File: Leetcode/565/565.go
const MAXN int = 1e5+5
var vis = [MAXN]bool{}

func dfs(nums []int, cycleLength int, index int, ans *int) {
	if !vis[nums[index]] {
		vis[nums[index]] = true
		dfs(nums, cycleLength+1, nums[index], ans)
	} else {
		if cycleLength > *ans {
			*ans = cycleLength
		}
	}
}
func arrayNesting(nums []int) int {
	ans := 1
    for i := 0; i < MAXN; i++ {
        vis[i] = false
    }
	for i := 0; i < len(nums); i++ {
		if !vis[i] {
			vis[i] = true
			dfs(nums, 1, i, &ans)
		}
	}
	return ans
}

File: Leetcode/565/565.cpp
constexpr int MAXN = 1e5+5;

class Solution {
public:
    int maxCycleLength;
    bitset<MAXN> vis;
    void dfs(vector<int>&nums, int cycleLength, int index) {
        if (!vis[nums[index]]) {
            vis[nums[index]] = true;
            dfs(nums, cycleLength+1, nums[index]);
        }
        else {
            maxCycleLength = max(maxCycleLength, cycleLength);
        }
    }
    int arrayNesting(vector<int>& nums) {
        maxCycleLength = 0;
        vis.reset();
        int n = (int)nums.size();
        for (int i = 0; i < n; ++i) {
            if (!vis[i]) {
                vis[i] = true;
                dfs(nums, 1, i);
            }
        }
        return maxCycleLength;
    }
};